hudi-hadoop-mr/src/main/java/org/apache/hudi/hadoop/realtime/RealtimeSplit.java

```diff
@@ -70,13 +78,24 @@
    */
   void setBasePath(String basePath);
 
+  void setHoodieVirtualKeyInfoOpt(Option<HoodieVirtualKeyInfo> hoodieVirtualKeyInfoOpt);
+
   default void writeToOutput(DataOutput out) throws IOException {
     InputSplitUtils.writeString(getBasePath(), out);
     InputSplitUtils.writeString(getMaxCommitTime(), out);
     out.writeInt(getDeltaLogPaths().size());
     for (String logFilePath : getDeltaLogPaths()) {
       InputSplitUtils.writeString(logFilePath, out);
     }
+    if (!getHoodieVirtualKeyInfoOpt().isPresent()) {
```

Comment:
Option.ifPresent.orElse() ? lets make use of the Option APIs!

Reply:
ifPresent returns avoid. will try using .map().orElse()

Reply:
I could not find any other ways to do this. We are not mapping to anything. We are just doing X or Y depending on whether its present or not. So, we can't use .map().orElse(). I checked other apis in Option, but could not find any.